src/net_processing.cpp

```diff
@@ -1514,7 +1511,9 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
         if (nVersion < MIN_PEER_PROTO_VERSION)
         {
             // disconnect from peers older than this proto version
-            LogPrintf("peer=%d using obsolete version %i; disconnecting\n", pfrom->GetId(), nVersion);
+            std::string obsolete_log(strprintf("peer=%d using obsolete version %i; disconnecting\n", pfrom->GetId(), nVersion));
+            if (pfrom->fInbound) LogPrint(BCLog::NET, "%s", obsolete_log);
+            else                 LogPrintf(           "%s", obsolete_log);
```

Comment:
Would prefer to create a BCLog::ALWAYS category and use something like `pfrom->fInbound ? BCLog::NET: BCLog::ALWAYS` instead of repeating the log statements in this way.